plotly/tools.py

```diff
@@ -2093,14 +2088,717 @@ def get_streamline_arrows(self):
+
+        # Create ohlc
+        ohlc = tls.FigureFactory.create_ohlc(open_data, high_data,
+                                             low_data, close_data,
+                                             dates=dates)
+
+        fig = ohlc
+        fig['layout'].update(title = 'OHLC Chart')
+
+        py.plot(fig, filename='ohlc with title', validate=False)
+        ```
+
+        Example 4: Plot OHLC with Pandas
+        ```
+        import pandas.io.data as web
+        import pandas as pd
+        from datetime import datetime
+
+        # Get Data
+        start = datetime(2010, 1, 1)
+        end = datetime(2012, 1, 27)
+        df = web.DataReader("aapl", 'yahoo', start, end)
+
+        # Get Dates
+        datelist = (pd.date_range(datetime(2010, 1, 1),
+                    periods=len(df.Open)).tolist())
+        d=[]
+        for i in range(len(datelist)):
+            d.append(datelist[i].to_datetime())
+
+        # Make OHLC
+        ohlc_panda = tls.FigureFactory.create_ohlc(df.Open, df.High,
+                                                   df.Low, df.Close,
+                                                   dates=d)
+
+        # Plot
+        py.plot(ohlc_panda, filename='ohlc with pandas', validate=False)
+
+        ```
+        """
+        if dates:
+            TraceFactory.validate_equal_length(open, high, low, close, dates)
+        else:
+            TraceFactory.validate_equal_length(open, high, low, close)
+        FigureFactory.validate_ohlc(open, high, low, close, direction,
+                                    **kwargs)
+
+        (flat_increase_x,
+         flat_increase_y,
+         text_increase) = _OHLC(open, high, low, close, dates).get_increase()
+
+        ohlc_incr = dict(type='scatter',
+                         x=flat_increase_x,
+                         y=flat_increase_y,
+                         mode='lines',
+                         name='Increasing',
+                         showlegend=False,
+                         line=Line(color='rgb(44, 160, 44)'),
+                         text=text_increase,
+                         **kwargs)
+
+        (flat_decrease_x,
+         flat_decrease_y,
+         text_decrease) = _OHLC(open, high, low, close, dates).get_decrease()
+
+        ohlc_decr = dict(type='scatter',
+                         x=flat_decrease_x,
+                         y=flat_decrease_y,
+                         mode='lines',
+                         name='Decreasing',
+                         showlegend=False,
+                         line=Line(color='rgb(214, 39, 40)'),
+                         text=text_decrease,
+                         **kwargs)
+
+        layout = dict(xaxis=dict(zeroline=False))
+
+        if direction is 'increasing':
+            return dict(data=[ohlc_incr], layout=layout)
+        elif direction is 'decreasing':
+            return dict(data=[ohlc_decr], layout=layout)
+        elif direction is 'both':
```

Comment:
:cow2: consider something like this? Otherwise, we'd just be returning `None`, right?

```
if direction is 'increasing':
    # blah
elif direction is 'decreasing':
    # blah
elif direction is 'both':
    # blah
else:
    raise SomeException('blah')
```